src/sentry/api/endpoints/organization_events_spans_performance.py

```diff
@@ -621,33 +588,49 @@ def query_example_transactions(
                 Op.IN,
                 Function(
                     "tuple",
-                    [Function("tuple", [suspect.op, suspect.group]) for suspect in spans],
+                    [Function("tuple", [span.op, span.group])],
                 ),
             ),
         ]
     )
 
-    if len(spans) > 1:
-        # Hack: the limit by clause only allows columns but here we want to
-        # do a limitby on the two array joins. For the time being, directly
-        # do the limitby on the internal snuba name for the span group column
-        # but this should not be relied upon in production, and if two spans
-        # differ only by the span op, this will result in a incorrect query
-        builder.limitby = LimitBy(Column("_snuba_array_join_spans_group"), per_suspect)
-
     snql_query = builder.get_snql_query()
     results = raw_snql_query(snql_query, "api.organization-events-spans-performance-examples")
 
-    examples: Dict[Span, List[EventID]] = {Span(suspect.op, suspect.group): [] for suspect in spans}
+    examples: Dict[Span, List[EventID]] = {Span(span.op, span.group): []}
 
     for example in results["data"]:
         key = Span(example["array_join_spans_op"], example["array_join_spans_group"])
-        value = EventID(example["project.id"], example["project"], example["id"])
+        value = EventID(params["project_id"][0], example["id"])
         examples[key].append(value)
 
     return examples
 
 
+def get_span_description(
+    event: EventID,
+    span_op: str,
+    span_group: str,
+    cache: Dict[EventID, Any],
+) -> Optional[str]:
+    if event not in cache:
+        cache[event] = eventstore.get_event_by_id(event.project_id, event.event_id)
```

Comment:
as per slack, I think get_event_by_id already caches so we don't need this extra cache